ui/app/AppLayouts/Wallet/views/CollectiblesView.qml

```diff
@@ -124,26 +123,43 @@ ColumnLayout {
         readonly property var nwFilters: root.networkFilters.split(":")
         readonly property var addrFilters: root.addressFilters.split(":").map((addr) => addr.toLowerCase())
 
-        function containsAnyAddress(ownership, filterList) {
-            for (let i = 0; i < ownership.count; i++) {
-                let accountAddress = ModelUtils.get(ownership, i, "accountAddress").toLowerCase()
-                if (filterList.includes(accountAddress)) {
-                    return true
+        function getLatestTimestmap(ownership, filterList) {
+            let latest = 0
+
+            if (!!ownership) {
+                for (let i = 0; i < ownership.count; i++) {
+                    let accountAddress = ModelUtils.get(ownership, i, "accountAddress").toLowerCase()
+                    if (filterList.includes(accountAddress)) {
+                        let txTimestamp = ModelUtils.get(ownership, i, "txTimestamp")
+                        latest = Math.max(latest, txTimestamp)
+                    }
                 }
             }
-            return false
+            return latest
         }
 
-        function getLatestTimestmap(ownership, filterList) {
-            let latest = 0
-            for (let i = 0; i < ownership.count; i++) {
-                let accountAddress = ModelUtils.get(ownership, i, "accountAddress").toLowerCase()
-                if (filterList.includes(accountAddress)) {
-                    let txTimestamp = ModelUtils.get(ownership, i, "txTimestamp")
-                    latest = Math.max(latest, txTimestamp)
+        function getBalance(ownership, filterList) {
+            // Balance is a Uint256, so we need to use AmountsArithmetic to handle it
+            let balance = AmountsArithmetic.fromNumber(0)
+
+            if (!!ownership) {
+                for (let i = 0; i < ownership.count; i++) {
+                    let accountAddress = ModelUtils.get(ownership, i, "accountAddress").toLowerCase()
+                    if (filterList.includes(accountAddress)) {
+                        let tokenBalanceStr = ModelUtils.get(ownership, i, "balance")+""
+                        if (tokenBalanceStr !== "") {
+                            let tokenBalance = AmountsArithmetic.fromString(tokenBalanceStr)
+                            balance = AmountsArithmetic.sum(balance, tokenBalance)
+                        }
+                    }
+                }
+                // For simplicity, we limit the result to the maximum int manageable by QML
+                const maxInt = 2147483647
```

Comment:
I think max is "99+"

Reply:
@caybro that's handled in`CollectibleBalanceTag.qml`, I wanted to keep the balance as raw as possible here in case we need it for some other purpose

Reply:
One thing to note, JS `Number.MAX_SAFE_INTEGER` is more than QML's `property int` can handle

Reply:
@IvanBelyakoff I wanted to insert the int limitation as late as possible... I don't want to have the nim model handle the selected addresses and calculate the resulting balance, so the next best thing I could quickly do is handle it here, which forces me to work with u256s. With this process I could have this int-limited value -useful for comparisons and further calculations- and the "true" balance in string representation in case I wanted to display that somewhere.

We can potentially make this all go away with https://github.com/status-im/status-desktop/issues/13880  


We could for example have the "real" value for display purposes, for which 


 If I didn't want to deal with big ints here I'd need to limit the values at the